docs/intro/pop.rst

```diff
@@ -0,0 +1,175 @@
+    |       ├── products.py
+    |       └── product_listings.py
+    └── setup.py
+
+Requirements for POP
+--------------------
+
+Minimum Requirements
+~~~~~~~~~~~~~~~~~~~~
+
+This covers the basic use case:
+
+    - Installation of **POP** either from public or private repositories:
+  
+      - PyPI
+      - Git
+
+    - Version specifiers that can be used to accommodate the various parser patches
+      that come along in any web data extraction project.
+    - Importing the Page Objects directly from the installed package in a project.
+
+
+This translates into **POPs** needing to have:
+
+    - The ``setup.py`` script which is the standard way of distributing Python packages.
+
+Thus, the most basic way of packaging **POPs** would be:
+
+.. code-block:: python
+
+    from setuptools import setup, find_packages
+
+    setup(
+        name='ecommerce-page-objects',
+        version='1.0.0',
+        packages=find_packages(),
+        install_requires=["web-poet"]
+    )
+
+This allows the **POP** to be installable via ``pip install ecommerce-page-objects==1.0.0``
+`(assuming it's deployed in PyPI)` or via a Git repo like 
+``pip install git+https://github.com/some-org/ecommerce-page-objects.git@1.0.0``
+`(assuming the repo is public)`.
+
+After installing the **POP**, anyone could access the Page Objects in it
+by simply importing them:
+
+.. code-block:: python
+
+    from ecommerce_page_objects.furniture_shop.products import FurnitureProductPage
+
+    response_data = download_response("https://www.furnitureshop.com/product/xyz")
+    page = FurnitureProductPage()
+    item = page.to_item()
+
+Recommended Requirements
+~~~~~~~~~~~~~~~~~~~~~~~~
+
+This covers these use use cases:
+
+    - The minimum requirements and use cases stated above
+    - The ability to retrieve the declared :class:`~.OverrideRule`
+      inside the **POP**
+
+This means that a list of :class:`~.OverrideRule` must be explicitly
+declared in the **POP**. This enables projects using the **POP** to know:
+
+    - which URL Patterns a given Page Object is expected to work
+    - what it's trying to override `(or replace)`
+
+This could be done by declaring a ``RULES`` variable that can be
+imported as a top-level variable from the package.
+
+For example, suppose our project is named **ecommerce_page_objects**
+and is using either of the project structure options discussed in the
+previous sections, then we can define the ``RULES`` as the following
+inside ``ecommerce_page_objects/ecommerce_page_objects/__init__.py``.
+
+.. code-block:: python
+
+    from web_poet import default_registry, consume_modules
+
+    consume_modules("ecommerce_page_objects")
+    RULES = default_registry.get_overrides()
+
+This allows any developer using a **POP** to easily get the list of
+:class:`~.OverrideRule` using the convention of accessing it via the
+``RULES`` variable as a top-level variable:
+
+.. code-block:: python
+
+    from ecommerce_page_objects import RULES
```

Comment:
When deploying a **POP** behind an API, the separate deployment CLI-tool can have a configuration file that points to this entry point variable.

For example, such config file could be named `pop.cfg` which is basically:

```
[deploy]
rules = ecommerce_page_objects.RULES

[projects]
dev = 123
prod = 7890
```